aptos-move/block-executor/src/executor.rs

```diff
@@ -610,14 +667,34 @@ where
                     {
                         data_map.write(key, write_op);
                     }
+                    for (group_key, group_ops) in output.resource_group_inner_ops().into_iter() {
+                        for (value_tag, group_op) in group_ops.into_iter() {
+                            data_map.insert_group_op(&group_key, value_tag, group_op);
+                        }
+                    }
+
                     // Calculating the accumulated gas costs of the committed txns.
                     let fee_statement = output.fee_statement();
                     accumulated_fee_statement.add_fee_statement(&fee_statement);
                     counters::update_sequential_txn_gas_counters(&fee_statement);
 
+                    let mut finalized_groups = vec![];
+                    for (group_key, metadata_op) in
+                        output.resource_group_write_metadata().into_iter()
+                    {
+                        let finalized_group = data_map.group_to_commit(&group_key);
+                        assert_eq!(
+                            finalized_group.is_empty(),
+                            metadata_op.is_deletion(),
+                            "Group contents not consistent with op kind"
+                        );
+                        finalized_groups.push((group_key, metadata_op, finalized_group));
+                    }
+
                     // No delta writes are needed for sequential execution.
-                    output.incorporate_delta_writes(vec![]);
-                    //
+                    output.record_groups(finalized_groups);
+                    output.incorporate_additional_writes(vec![]);
```

Comment:
doesn't the order matter?


we first need to incorporate delta writes, and do aggv2 exchange, and only then do resource group merging?

Reply:
delta writes are for V1, and incorporate delta writes simply adds recorded groups and V1 writes to the TransactionOutput write set. The exchange would indeed have to occur before these lines.